File: MythonC++/Mython/NameException.cpp
#include "NameException.h"

NameException::NameException(string varName) : exception()
{
    this->errorMessage = "NameError: name \'" + varName + "\' is not defined";
}

const char* NameException::what() const throw()
{

    return this->errorMessage.c_str();
}

File: MythonJava/src/sample/exceptions/NameException.java
package sample.exceptions;

public class NameException extends InterpreterException {

    private String errorMessage;

    public NameException(String name) {
        this.errorMessage = "Name error: name \'" + name + "\' is not defined";
    }

    @Override
    public void printStackTrace() {
        System.out.println(ANSI_RED + this.errorMessage + ANSI_RESET);
    }
}
